src/Raven.Server/Documents/Queries/IndexQueryServerSide.cs

```diff
@@ -305,5 +339,168 @@ private static void AssertPaging(IndexQueryServerSide indexQuery)
+                {
+                    throw new InvalidQueryException("Collection query does not support filtering by id() using Between operator. Supported operators are: =, IN",
+                        QueryText, parameters);
+                }
+            }
+
+            public override void VisitIn(QueryExpression fieldName, List<QueryExpression> values, BlittableJsonReaderObject parameters)
+            {
+                // Handles the case of IN with empty list
+                Ids ??= new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+                if (fieldName is MethodExpression me && string.Equals("id", me.Name.Value, StringComparison.OrdinalIgnoreCase))
+                {
+                    foreach (var item in values)
+                    {
+                        if (item is ValueExpression iv)
+                        {
+                            foreach (var id in QueryBuilder.GetValues(_query, _metadata, parameters, iv))
+                            {
+                                AddId(id.Value?.ToString());
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    throw new InvalidQueryException("Collection query does not support filtering by id() using Between operator. Supported operators are: =, IN",
+                        QueryText, parameters);
+                }
+            }
+
+            public override void VisitMethodTokens(StringSegment name, List<QueryExpression> arguments, BlittableJsonReaderObject parameters)
+            {
+                var expression = arguments[^1];
+                if (expression is BinaryExpression be && be.Operator == OperatorType.Equal)
+                {
+                    VisitFieldToken(new MethodExpression("id", new List<QueryExpression>()), be.Right, parameters, be.Operator);
+                }
+                else if (expression is InExpression ie)
+                {
+                    VisitIn(new MethodExpression("id", new List<QueryExpression>()), ie.Values, parameters);
+                }
+                else if (string.Equals(name.Value, "startsWith", StringComparison.OrdinalIgnoreCase))
+                {
+                    if (expression is ValueExpression iv)
+                    {
+                        var prefix = QueryBuilder.GetValue(_query, _metadata, parameters, iv);
+                        StartsWith = prefix.Value?.ToString();
+                    }
+                }
+                else
+                {
+                    ThrowNotSupportedCollectionQueryOperator(expression.Type.ToString(), parameters);
+                }
+            }
+
+            private void AddId(string id)
+            {
+                Ids ??= new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
this is wrong, the storage method is case sensitive
also this needs to be lowercased Slice IMO as it was

Reply:
reverted